### ISSUE TITLE
Periods in function name are seen as namespace/module names in Visual Studio's Test Explorer (F# unit test, V3)

### ISSUE THREAD

**Issue by richardcox13** [xunit/xunit#3013]
Given this `UnitRwsr1.fs` in a Xunit V3 F# project:

```fsharp
module Demo.UnitTest1

open Xunit

[<Fact>]
let ``This.is.a.valid.name.in.F#`` () =
    Assert.True(true)
```

Then see this in Visual Studio's (2022 17.12.0.Preview 1.0);

![image](https://github.com/user-attachments/assets/89fea834-bd29-44f5-9326-9af7839fe28b)

where it is clear the periods in the function name have been parsed as part of the namespace and module. (Without periods in the function name Test Explorer displays correctly.)

(It is not clear to me where the unit test runner to VS integration point is, so I will understand if this is a VS issue...)

**Comment by richardcox13** [comment]
Thanks. As expected :-( !

I've raised a VS Bug https://developercommunity.visualstudio.com/t/Test-Explorer-Fails-to-Correctly-Display/10733339?port=1025&fsid=6c2b1158-1717-4a6c-b5df-902ac7c8040e**** but really expecting to wait on the `Microsoft.Test.Platform` support...

**Comment by bradwilson** [comment]
> I've raised a VS Bug

I'm guessing they're going to say "won't fix" because of Microsoft.Testing.Platform being available soon, but worth a shot.

> really expecting to wait on the `Microsoft.Test.Platform` support...

Their tentative ship date is sometime in September for the M.T.P binaries, but I have no idea when the RTM version of VS 17.12 will be. Since you're already using the pre-release, then you won't have to wait for that. 😄 If you continue to include `xunit.runner.visualstudio` then you'll have back-compat with older runners, so in case your team is not using VS pre-release, they'll continue to get Test Explorer support, but with the incorrect names.

**Comment by richardcox13** [comment]
Indeed.

I predict VS 17.12 RTM will, based on the last few years, will be close to .NET 9 RTW.

**Comment by richardcox13** [comment]
Retesting with updated XunitV3 dependencies and enabling Microsoft.Testing.Platform and this is fixed! 

Not sure I'm a fan of the prefixing of each fact name (or theory instance) it its immediately containing class or module, but that's not a F# only thing.

**Comment by bradwilson** [comment]
Yeah, we have less control over what's displayed in Test Explorer when running in Microsoft Testing Platform mode. It appears, for the moment, to ignore the display name of the test and just do straight up sorting/display based on class name.

**Comment by bradwilson** [comment]
Closing as "external".
